File: LLM-Translations/GPT-4/avatar/Python/Java/codeforces_268_A.java
import java.util.ArrayList;
import java.util.Scanner;

public class codeforces_268_A {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.nextLine();
        ArrayList<String[]> q = new ArrayList<>();
        int count = 0;

        for (int i = 0; i < n; i++) {
            String[] a = sc.nextLine().split(" ");
            q.add(a);
        }

        for (String[] j : q) {
            for (String[] k : q) {
                if (k == j) {
                    continue;
                } else if (j[0].equals(k[k.length - 1])) {
                    count++;
                } else {
                    continue;
                }
            }
        }

        System.out.println(count);
    }
} // End of Code

File: LLM-Translations/GPT-4/avatar/Python/C/codeforces_268_A.c
#include <stdio.h>
#include <string.h>

int main() {
    int n, count = 0;
    scanf("%d", &n);
    char q[n][20];

    for(int i = 0; i < n; i++) {
        scanf("%s", q[i]);
    }

    for(int j = 0; j < n; j++) {
        for(int k = 0; k < n; k++) {
            if(k == j) {
                continue;
            }
            else if(q[j][0] == q[k][strlen(q[k]) - 1]) {
                count++;
            }
            else {
                continue;
            }
        }
    }

    printf("%d", count);

    return 0;
} 
// End of Code
